block/manager.go

```diff
@@ -170,7 +170,24 @@ func (m *Manager) SetDALC(dalc da.DataAvailabilityLayerClient) {
 
 // AggregationLoop is responsible for aggregating transactions into rollup-blocks.
 func (m *Manager) AggregationLoop(ctx context.Context) {
+	initialHeight := uint64(m.genesis.InitialHeight)
+	height := m.store.Height()
+	var delay time.Duration
+
+	// TODO(tzdybal): double-check when https://github.com/celestiaorg/rollmint/issues/699 is resolved
+	if height < initialHeight {
+		delay = time.Until(m.genesis.GenesisTime)
+	} else if height >= initialHeight {
+		delay = time.Until(m.lastState.LastBlockTime.Add(m.conf.BlockTime))
+	}
+
+	if delay > 0 {
+		m.logger.Info("Waiting to produce block", "delay", delay)
+		time.Sleep(delay)
+	}
```

Comment:
I feel like the case `height < initialHeight` should produce an error as if I understand the code correctly it means that the given height in manager is smaller then the height initialised in genesis, which in turn means that the height in manager is too small, which should not happen. Maybe we can move this check while initialising the manager.

Reply:
Maybe this is related to #699

Reply:
Height in the store might be smaller than initial height in genesis - this is the case before first produced block - store will return 0. Also see: https://github.com/celestiaorg/rollmint/blob/8ac600796c2d61979ad44ca47c15af2ac8711e13/block/manager.go#L444-L445.

This should be revisited during or after #699.